### PR TITLE
feat: add --createissue flag, UT, bin/dev, fixed linked plugin check

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
adds `--createissue` flag to create a GH issue from the CLI
fixes the check for if a plugin is linked

### What issues does this PR fix or reference?
@W-11726998@

### CODE REVIEW

File: src/commands/doctor.ts [hunk 1]
```diff
@@ -105,9 +121,73 @@ export default class Doctor extends SfdxCommand {
...
     diagnosis.suggestions.forEach((s) => this.ux.log(`  * ${s}`));
 
+    if (createissueFlag) {
+      const raw = 'https://raw.githubusercontent.com/forcedotcom/cli/main/.github/ISSUE_TEMPLATE/bug_report.md';
+      const ghIssue = await got(raw, {
+        throwHttpErrors: false,
+        agent: { https: ProxyAgent(getProxyForUrl(raw)) },
+      });
+
+      const title = await this.ux.prompt('Please enter a title for the issue');
```
Comment (jshackell-sfdc) [inline src/commands/doctor.ts:131]:
```suggestion
      const title = await this.ux.prompt('Enter a title for your new issue');
```

File: messages/doctor.js [hunk 2]
```diff
@@ -28,5 +29,5 @@ Run CLI doctor diagnostics for a specific plugin:
   doctorNotInitializedError: 'Must first initialize a new SfDoctor.',
   doctorAlreadyInitializedError: 'SfDoctor has already been initialized.',
   pluginNotInstalledError:
-    'Specified plugin [%s] isn\'t installed. Install it, correct the name, or choose another plugin.',
+    "Specified plugin [%s] isn't installed. Install it, correct the name, or choose another plugin.",
```
Comment (shetzel) [inline messages/doctor.js:32]: Doesn't this fail linting for using double quotes?

File: src/commands/doctor.ts [hunk 3]
```diff
@@ -45,13 +54,23 @@ export default class Doctor extends SfdxCommand {
...
   private filesWrittenMsgs: string[] = [];
 
+  /**
+   * Only made into its own method for unit testing purposes
+   *
+   * @param url: url string to open
+   */
+  public static async openUrl(url: string): Promise<void> {
+    await open(url);
+  }
```
Comment (shetzel) [inline src/commands/doctor.ts:64]: What do you think about making this a private method and stubbing it in the test?  I know you'll have to add a `ts-ignore` and an eslint-ignore but I think that's fine to have in a test.

File: messages/doctor.js (hunk 2 above)
Comment (WillieRuemmele) [reply messages/doctor.js:32]: not sure what was going on there, I'm not seeing any linting errors, reverting it

**Review by shetzel — APPROVED** [review]

**Pull request merged**